internal/handler/job/events/events.go

```diff
@@ -45,16 +54,19 @@ const (
 	ActionUpdateDefinition Action = "UPDATE_DEFINITION"
 )
 
-// NOTE: The number 32 for capacity is arbitrary; suggestions for a different value are welcome, particularly if supported by evidence.
-var e *emitter.Emitter = emitter.New(32)
+var (
+	subscribers = make([]Subscriber, 0)
+	mu          sync.RWMutex
+)
+
+// how many messages are kept in the subscriber's backlog before the subscriber is removed due to being considered dead
+const backlogCapacity int = 32
```

Comment:
Hm, why opting for a count instead of a time period here?

Reply:
This is the channel's buffer size, which is count-based, see https://go.dev/tour/concurrency/3.

Reply:
Still, isn't a time-based mechanism better? If not, why?

Reply:
Pushed a time-based approach now